packages/grid/src/InternalGrid.tsx

```diff
@@ -0,0 +1,312 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import React, {
+  useState,
+  useRef,
+  useEffect,
+  useCallback,
+  MutableRefObject,
+  useMemo,
+} from 'react';
+import {
+  Grid as VirtualisedGrid,
+  CellMeasurer,
+  ScrollPosition,
+  GridCellProps,
+} from 'react-virtualized';
+import { setIn } from 'utilitify';
+import {
+  Body,
+  BodyCell,
+  BodyCellOffset,
+  BodyCellContent,
+  ExpandButtonWrapper,
+  Wrapper,
+  TotalBlock,
+  TotalCellContent,
+  TotalCell,
+  ShiftInsteadButton,
+  TotalsShift,
+} from './styled';
+import { AddIcon, RemoveIcon } from './icons';
+import { searchLastVisible } from './utils';
+
+import { HeaderWrapper } from './HeaderWrapper';
+import {
+  IColumn, IInternalGrid,
+} from './interfaces';
+
+
+const InternalGrid = ({
+  rightScroll = true,
+  bottomScroll = true,
+  width = 0,
+  height = 0,
+  shouldMovingColumns = true,
+  shouldChangeColumnsWidth = true,
+  scrollTop,
+  onScrollsyncScroll,
+  gridHOCMappedColumns,
+  setGridHOCMappedColumns,
+  resizeGridAfterResizeLastColumn,
+  gridPosition,
+  onChangeColumns = () => {},
+  totals,
+  handleSelect,
+  onChangeExpand,
+  mappedItems = [],
+  selectedRows,
+  cacheRef,
+  themeRef,
+  rowHeight,
+  isScrollingOptOut,
+  overscanColumnCount,
+  overscanRowCount,
+}: IInternalGrid) => {
+  const [mappedColumns, setMappedColumns] = useState<IColumn[]>(gridHOCMappedColumns);
+  const fullWidthRef = useRef(
+    mappedColumns.filter(({ visible }: IColumn) => visible).reduce(
```

Comment:
You use the same code in Grid.txs
```
leftMappedColumns
  .filter(({ visible }: IColumn) => visible)
  .reduce((acc, { width: $width }) => Number(acc) + Number($width), 0),
```

```
rightMappedColumns
  .filter(({ visible }: IColumn) => visible)
  .reduce((acc, { width: $width }) => Number(acc) + Number($width), 0),
```
I think it is better create some general function or do it only in one place.
Check this moment please.

Reply:
no, inside the Grid it is used for calculating the size of fixed grids, and inside the grid it is used for headers